src/main/groovy/ru/endlesscode/bukkitgradle/meta/extension/MetaItem.groovy

```diff
@@ -31,8 +32,12 @@ class MetaItem {
      * Validates meta item
      *
      * @return true if meta item valid, otherwise false
+     * @throws GradleException when item invalid
      */
-    boolean isValid() {
-        return !this.required || getValue() != null
+    boolean validate() {
+        if (this.required && getValue() == null) {
+            throw new GradleException("Plugin metadata parse error: '$id' must not be null")
```

Comment:
I think it's better to return boolean as it was before and throw an exception in place of the call only if needed.

Reply:
I think throwing exception inside object is more OOP-style because in this case you can throw exceptions with different messages if you have more complex validation and validation logic does not leaks outside.

Reply:
And example of this is https://github.com/EndlessCodeGroup/BukkitGradle/blob/ed01fb6378ef4b73c2d086a4daf8f1001f80986d/src/main/groovy/ru/endlesscode/bukkitgradle/meta/extension/EnumeratedMetaItem.groovy

Reply:
I figured out your idea.
It is not a best-possible approach but it is OK. I think this method should not return boolean then.